src/AccessibilityInsights.Rules/Library/ControlShouldNotSupportValuePattern.cs

```diff
@@ -22,15 +23,19 @@ public ControlShouldNotSupportValuePattern()
         public override EvaluationCode Evaluate(IA11yElement e)
         {
             if (e == null) throw new ArgumentException(nameof(e));
+            
+            var name = e.Name.Trim().ToLower();
+            var value = e.GetPattern(PatternType.UIA_ValuePatternId).GetValue<string>("Value").Trim().ToLower();
```

Comment:
Even though you have already checked to make sure the value pattern exists and the value is not null as part of the condition, it may not be a bad idea to confirm they are not null here. My concern is that the condition could easily change without the developer realizing the evaluation depends on those checks.